### ISSUE TITLE
Datagrid focus management 

### ISSUE THREAD

**Issue by gnomeontherun** [vmware/clarity#3370]
Inside of the Datagrid, there are a few interactions that the focus is not managed in the best way.

* [x] When using the page count select box, when the selection is changed the focus should move back to the start of the newly displayed content.
* [x] When using the previous/next pagination buttons, the focus should move back to the start of the newly displayed content.
* [x] When closing a filter, the focus is lost to the page when it should return to the filter toggle button.
* [x] When using the select all button, the focus is lost to the page when it should remain on the select all button.
* [x] Screen reader users will not be aware of actions available from the clr-dg-action-overflow component when it appears because focus remains on the trigger but should move into the menu.

**Comment by gnomeontherun** [comment]
@Jinnie I added one more task above from #3371 since it is part of focus management.

**Comment by Jinnie** [comment]
> When using the select all button, the focus is lost to the page when it should remain on the select all button.

Is this about the hideable columns `select all`? It gets disabled after all options are selected. We can't give focus to a disabled form item. @gnomeontherun, please provide more details about this sub-task.

**Comment by Jinnie** [comment]
Update about "Select All" button topic - after an offline discussion it was decided that focus should go to the header and a success message should be announced.

**Comment by Jinnie** [comment]
Short update about the Hideable / Select All subtask:
* [ ] Focus lost on column checkbox toggle
* [ ] Nothing announced after dialog opened
